File: animals war.py
import random
class Pos:
    def __init__(self,x,y):
        self.x=x
        self.y=y
class Mohreh:
    def __init__(self,name,power,distance,x,y):
        self.name=name
        self.power=power
        self.distance=distance
        self.pos=Pos(x,y)
class Bazikon:
    def __init__(self,name_e,power_e,distance_e,pos_e_x,pos_e_y,name_h,power_h,distance_h,pos_h_x,pos_h_y,name_t,power_t,distance_t,pos_t_x,pos_t_y,number):
        self.m_e=Mohreh(name_e,power_e,distance_e,pos_e_x,pos_e_y)
        self.m_h=Mohreh(name_h,power_h,distance_h,pos_h_x,pos_h_y)
        self.m_t=Mohreh(name_t,power_t,distance_t,pos_t_x,pos_t_y)
        self.number=number
    def change_dir(self,mohreh,direction,distance):
        if(direction=="u"):
            m_pos_x=mohreh.pos.x-distance
            m_pos_y=mohreh.pos.y
        elif(direction=="d"):
            m_pos_x=mohreh.pos.x+distance
            m_pos_y=mohreh.pos.y
        elif(direction=="r"):
            m_pos_x=mohreh.pos.x
            m_pos_y=mohreh.pos.y+distance
        else:
            m_pos_x=mohreh.pos.x
            m_pos_y=mohreh.pos.y-distance
        return (m_pos_x,m_pos_y)
    def search(self,mohreh,direction,distance,z_bazi):
        tu_ch=self.change_dir(mohreh,direction,distance)
        if(z_bazi[tu_ch[0]][tu_ch[1]]==None):
            return True
        else:
            if(z_bazi[tu_ch[0]][tu_ch[1]].name.isupper() == mohreh.name.isupper() or z_bazi[tu_ch[0]][tu_ch[1]].name.islower() == mohreh.name.islower()):
                return False
            elif(z_bazi[tu_ch[0]][tu_ch[1]].name.isupper() != mohreh.name.isupper() and z_bazi[tu_ch[0]][tu_ch[1]].power<=mohreh.power):
                return True
            else:
                return False
    def move(self,mohreh,direction,distance,z_bazi):
        tu_ch=self.change_dir(mohreh,direction,distance)
        if(self.search(mohreh,direction,distance,z_bazi)==True):
            z_bazi[tu_ch[0]][tu_ch[1]]=mohreh
            z_bazi[mohreh.pos.x][mohreh.pos.y]=None
            mohreh.pos.x=tu_ch[0]
            mohreh.pos.y=tu_ch[1]
        else:
            print("harekat gheyr mojaz!!")
bazikon_1=Bazikon("e",2,2,0,0,"h",1,1,0,1,"t",3,1,0,2,1)
bazikon_2=Bazikon("E",2,2,2,0,"H",1,1,2,1,"T",3,1,2,2,2)
zamin_b=[[bazikon_1.m_e,bazikon_1.m_h,bazikon_1.m_t],[None,None,None],[bazikon_2.m_e,bazikon_2.m_h,bazikon_2.m_t]]
def print_z(zamin_b):
    for i in range(len(zamin_b)):
        for j in range(len(zamin_b[0])):
            if(zamin_b[i][j]==None):
                print("*\t",end="")
            else:
                print(zamin_b[i][j].name+"\t",end="")
        print()
def interface_b(bazikon):
    if(bazikon.number==1):
        while 1:
            mohreh=input("Player 1 choose mohre(e h t):")
            direction=input("Choose direction: (L U R D) : (Left , Right , Up , Bottom):")
            distance=int(input("Choose Distance:"))
            if((direction=="l" or direction=="r" or direction=="d" or direction=="u")):
                if(mohreh=="e" and distance<=bazikon.m_e.distance):
                    break
                elif(mohreh=="h" and distance<=bazikon.m_h.distance):
                    break
                elif(mohreh=="t" and distance<=bazikon.m_t.distance):
                    break
                else:
                    print("lotfan etelaat ra sahih vared konid!!!!")
                    pass
    else:
        while 1:
            mohreh=input("Player 2 choose mohre(E H T):")
            direction=input("Choose direction: (L U R D) : (Left , Right , Up , Bottom):")
            distance=int(input("Choose Distance:"))
            if((direction=="l" or direction=="r" or direction=="d" or direction=="u")):
                if(mohreh=="E" and distance<=bazikon.m_e.distance):
                    break
                elif(mohreh=="H" and distance<=bazikon.m_h.distance):
                    break
                elif(mohreh=="T" and distance<=bazikon.m_t.distance):
                    break
                else:
                    print("lotfan etelaat ra sahih vared konid!!!!")
                    pass
    return mohreh,direction,distance
def search_end(zamin_b):
    lower=0
    upper=0
    for i in range(len(zamin_b)):
        for j in range(len(zamin_b[0])):
            if(zamin_b[i][j]!=None):
                if(zamin_b[i][j].name.isupper()==True):
                    upper+=1
                else:
                    lower+=1
    if(upper!=0 and lower==0):
        return 1
    elif(upper==0 and lower!=0):
        return 0
    else:
        return -1
def run_bazi(bazikon_1,bazikon_2,zamin_b):
    if(random.randrange(10)%2!=0):
        tu_karbar=interface_b(bazikon_1)
        mohreh=None
        if(tu_karbar[0]=="e"):
            mohreh=bazikon_1.m_e
        elif(tu_karbar[0]=="h"):
            mohreh=bazikon_1.m_h
        else:
            mohreh=bazikon_1.m_t
        bazikon_1.move(mohreh,tu_karbar[1],tu_karbar[2],zamin_b)
        print_z(zamin_b)
    else:
        tu_karbar=interface_b(bazikon_2)
        mohreh=None
        if(tu_karbar[0]=="E"):
            mohreh=bazikon_2.m_e
        elif(tu_karbar[0]=="H"):
            mohreh=bazikon_2.m_h
        else:
            mohreh=bazikon_2.m_t
        bazikon_2.move(mohreh,tu_karbar[1],tu_karbar[2],zamin_b)
        print_z(zamin_b)
print_z(zamin_b)
while 1:
    run_bazi(bazikon_1,bazikon_2,zamin_b)
    if(search_end(zamin_b)==1):
        print("bazikon 2 winnnnnnnner!")
        break
    elif(search_end(zamin_b)==0):
        print("bazikon1 winnnnnnnnner!")
        break
    else:
        pass

File: animals war.java
import java.util.Scanner;
import java.lang.Math;
class Pos
{
    int x;
    int y;
}
class Mohre
{
    Pos pos=new Pos();
    int pow;
    int dis;
    char name;
    Mohre(int pow,int dis,char name,int x,int y)
    {  
      this.pow=pow;
      this.dis=dis;
      this.name=name;
      this.pos.x=x;
      this.pos.y=y;
    }
}
class Bazikon
{   
    Mohre tiger,elephant,horse;
    Bazikon(int d,int d1,int d2,int p,int p1,int p2,char name,char name1,char name2,int x,int x1,int x2,int y,int y1,int y2)
    {
        tiger=new Mohre(p,d,name,x,y);
        elephant=new Mohre(p1,d1,name1,x1,y1);
        horse=new Mohre(p2,d2,name2,x2,y2);
    }
    int search(Mohre b,int x1,int y1,Mohre z[][])
    {
        int i;
        if(b.name>='a'&&b.name<='z')
            i=1;
        else
            i=0;
        int j=0;
        if(z[x1][y1]!=null)
        {
            if(z[x1][y1].name>='a'&&z[x1][y1].name<='z')
                j=1;
            else
                j=0;
        }
        if(z[x1][y1]==null)    
            return 0;
        else
        {
            if((z[x1][y1].pow<=b.pow)&&(i!=j))                
                return 1;           
            else                            
                return -1;
        }
    }
    void harekat(char a,Mohre b,int l,Mohre z[][])
    {
        int x1=b.pos.x;
        int y1=b.pos.y;
        if(a=='u')
            x1-=l;
        else if(a=='d') 
            x1+=l;
        else if(a=='r')
            y1+=l;
        else if(a=='l')
            y1-=l;
        if((x1>=0&&x1<=2)&&(y1>=0&&y1<=2))
        {
            if(search(b,x1,y1,z)==-1)
                System.out.println("harekat gheyr mojaz!");
            else if(search(b,x1,y1,z)==0)
            {
                z[x1][y1]=b;
                z[b.pos.x][b.pos.y]=null;
                z[x1][y1].pos.x=x1;
                z[x1][y1].pos.y=y1;
            }
            else if(search(b,x1,y1,z)==1)
            {
                z[x1][y1]=b;
                z[b.pos.x][b.pos.y]=null;
                z[x1][y1].pos.x=x1;
                z[x1][y1].pos.y=y1;
            }
        }
        else
            System.out.println("waaaaarning!");      
    }
}
class Bazi
{
    Scanner input=new Scanner(System.in);
    Bazikon bazikon1=new Bazikon(1,2,1,3,2,1,'t','e','h',0,0,0,2,0,1);
    Bazikon bazikon2=new Bazikon(1,2,1,3,2,1,'T','E','H',2,2,2,2,0,1);
    Mohre[][]z={{bazikon1.elephant,bazikon1.horse,bazikon1.tiger},{null,null,null},{bazikon2.elephant,bazikon2.horse,bazikon2.tiger}};  
    void print()
    {
        int i,j;
        int k=0;
        for(i=0;i<z.length;i++)
        {
            for(j=0;j<z.length;j++)
            {
                if(z[i][j]!=null)
                {
                    System.out.print(z[i][j].name+"\t");
                    k++;
                    if(k%3==0)
                        System.out.print("\n");
                }
                else
                {
                     System.out.print("*\t");
                     k++;
                     if(k%3==0)
                        System.out.print("\n");
                }
            }
        }
    }
    int search()
    {
        int i,j;
        int k=0;
        int b=0;
        for (i = 0; i < z.length; i++) 
        {
            for (j = 0; j < z.length; j++)
            {
                if(z[i][j]!=null)
                {
                    if(z[i][j].name>='a'&&z[i][j].name<='z')
                        k++; 
                    else if(z[i][j].name>='A'&&z[i][j].name<='Z')
                        b++;
                }
            }
        }
        if(k==0&&b!=0)
        {
            System.out.println("player 2 winnnnnnner!");
            return 2;
        }
        else if(k!=0&&b==0)
        {
            System.out.println("player 1 winnnnnnner!");
            return 1;
        }   
        else
            return 0;
    } 
    void player1()
    {
        char n;
        while(true)
        {
            System.out.print("Player 1 choose mohre(e h t):\n");
            n=input.next().charAt(0);
            if(n=='e'||n=='h'||n=='t')
                break;
            else
                System.out.println("lotfan yeki az character haye mored nazar ra vared konid!");
        }
        System.out.println("Choose direction: (L U R D) :");
        char m=input.next().charAt(0);
        System.out.println("Choose Distance:");
        int l=input.nextInt();
        int i,j;
        boolean w=false;
        int k=0;
        int f=0;
        for(i=0;i<z.length;i++)
        {
            if(w==true)
                break;
            for(j=0;j<z.length;j++)
            {
                if(z[i][j]!=null)
                {
                    f++;
                    if(z[i][j].name==n)
                    {
                        if(z[i][j].dis>=l)
                        {
                            bazikon1.harekat(m,z[i][j],l,z);
                            w=true;
                            break;
                        }
                        else
                            System.out.println("distance gheyr mojaz!");
                    }
                    else
                      k++;  
                }
            }
        }
        if(f==k)
            System.out.println("mohre mored nazar vojod nadarad!");
        print();
    }
    void player2()
    {
        char n;
        while(true)
        {
            System.out.print("Player 2 choose mohre(E H T):\n");
            n=input.next().charAt(0);
            if(n=='E'||n=='H'||n=='T')
                break;
            else
                System.out.println("lotfan yeki az character haye mored nazar ra vared konid!");
        }
        System.out.println("Choose direction: (L U R D) :");
        char m=input.next().charAt(0);
        System.out.println("Choose Distance:");
        int l=input.nextInt();
        int i,j;
        boolean w=false;
        int k=0;
        int f=0;
        for(i=0;i<z.length;i++)
        {
            if(w==true)
                break;
            for(j=0;j<z.length;j++)
            {
                if(z[i][j]!=null)
                {
                    f++;
                    if(z[i][j].name==n)
                    {
                        if(z[i][j].dis>=l)
                        {
                            bazikon2.harekat(m,z[i][j],l,z);
                            w=true;
                            break;
                        }
                        else
                            System.out.println("distance gheyr mojaz!");
                    }
                    else
                        k++;
                }
            }
        }
        if(f==k)
            System.out.println("mohre mored nazar vojod nadarad!");
        print();
    }
    void random(int a)
    {
        if(a%2==0)
        {
            while(true)
            {
                player1();
                if(search()==1||search()==2)
                    break;
                player2();
                if(search()==1||search()==2)
                    break;
            }
        }
        else
        {
            while(true)
            {
                player2();
                if(search()==1||search()==2)
                    break;
                player1();
                if(search()==1||search()==2)
                    break;
            }
        }
    }
}
public class Main
{
    public static void main(String[] args) 
    {
       Bazi bazi=new Bazi();
       double a=Math.random()*10;
       int b=(int)a;
       bazi.print();
       bazi.random(b);
    }
}